test/unit/controller/register-user.controller.test.js

```diff
@@ -7,55 +7,61 @@ const userFixtures = require('../../fixtures/user.fixtures')
 const inviteFixtures = require('../../fixtures/invite.fixtures')
 const paths = require('../../../app/paths')
 const User = require('../../../app/models/User.class')
-const { RegistrationSessionMissingError } = require('../../../app/errors')
+const { RegistrationSessionMissingError, RESTClientError, ExpiredInviteError } = require('../../../app/errors')
+const { expect } = require('chai')
 
 describe('Register user controller', () => {
   const email = 'invited-user@example.com'
-  const code = 'a-code'
+  const inviteCode = 'a-code'
   const correlationId = 'a-request-id'
   const serviceExternalId = 'a-service-id'
 
+  const verifyOtpSuccessStub = sinon.spy(() => Promise.resolve())
   const completeInviteSuccessStub = sinon.stub().returns(inviteFixtures.validInviteCompleteResponse({
```

Comment:
This already existed, but I think it should return `Promise.resolve(InviteFixtures.validInviteCompleteResponse(...))` as the complete method returns a Promise. This will just ensure that we're awaiting it correctly in the tests - otherwise they'd still pass even if we didn't `await` in the controller.

Also looks like stubs and spies can be used interchangeably here. Would be good to use the same thing for both these spy/stubs just to avoid confusion. I guess using a stub is maybe technically more correct but we use spy in most places.

Reply:
Ah, I did want to do it the way you're suggesting and I went and looked for something like `InviteFixtures.validInviteCompleteResponse`, but originally failed to spot it.  I agree that it would be best to be consistent here, and now that I know what fixture I can return, I will change the verifyOtp one.

Reply:
Wait, hold on a sec, isn't `InviteFixtures.validInviteCompleteResponse` only appropriate for invite completion?  Surely what we need is something along the line of `validOtpVerifyResponse`?  Sorry, I misread your comment.

Reply:
I did indeed misread the comment, but not in the way I thought; Stephen has explained it to me on a call.